### PR TITLE
feat(javascript): improve on task trigger type guards

### PR DESCRIPTION
**Pull request by Fluf22**
## 🧭 What and Why

🎟 JIRA Ticket: Related to [DI-873](https://algolia.atlassian.net/browse/DI-873)

### Changes included:
- export type guard functions directly
- handle undefined trigger value

[DI-873]: https://algolia.atlassian.net/browse/DI-873?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by shortcuts — DISMISSED** [review]
looks good!

**Review by shortcuts — APPROVED** [review]
<3 nice catch!

File: templates/javascript/clients/client/api/guards.mustache [hunk 1]
```diff
@@ -5,33 +5,33 @@
...
  * @param trigger - The given Task Trigger.
  */
-isSubscriptionTrigger(
+export function isSubscriptionTrigger(
   trigger: Trigger
 ): trigger is SubscriptionTrigger {
   return trigger.type === 'subscription';
-},
+}
 {{/isIngestionClient}}
```
Comment (shortcuts) [inline templates/javascript/clients/client/api/guards.mustache:37]: just realized the check here but you don't need it in the main file in that case

**Review by shortcuts — APPROVED** [review]

File: templates/javascript/clients/client/api/guards.mustache (hunk 1 above)
Comment (Fluf22) [reply templates/javascript/clients/client/api/guards.mustache:37]: Well, I kept it for safety: in the future, if we want to add type guards for another client, we will surely remove the one in the main file, so this functions will already be protected.
Should I still remove it from one of the file?

Comment (shortcuts) [reply templates/javascript/clients/client/api/guards.mustache:37]: > Well, I kept it for safety: in the future, if we want to add type guards for another client, we will surely remove the one in the main file, so this functions will already be protected.

yup make sense it's a non blocking comment anyway there's no impact at all

feel free to remove it or not but it won't change anything in the end

**Pull request merged**